packages/core/src/config/config-resolvers.ts

```diff
@@ -411,7 +411,22 @@ function groupStyleguideAssertionRules({
   // Collect assertion rules
   const assertions: Assertion[] = [];
   for (const [ruleKey, rule] of Object.entries(rules)) {
-    if (ruleKey.startsWith('assert/') && typeof rule === 'object' && rule !== null) {
+    // keep the old assert/ syntax as an alias
+    if (
+      ( ruleKey.startsWith('assert/')) &&
+      typeof rule === 'object' &&
+      rule !== null
+    ) {
+    logger.warn(
+      `The 'assert/' syntax in ${ruleKey} is deprecated. Update your configuration to use 'rule/' instead. Examples and more information: https://redocly.com/docs/cli/rules/configurable-rules/\n`
+  );
+    }
+
+    if (
+      (ruleKey.startsWith('rule/') || ruleKey.startsWith('assert/')) &&
+      typeof rule === 'object' &&
+      rule !== null
+    ) {
       const assertion = rule as RawAssertion;
```

Comment:
I'd move the firs `if` clause into the second one like this:

```suggestion

    if (
      (ruleKey.startsWith('rule/') || ruleKey.startsWith('assert/')) &&
      typeof rule === 'object' &&
      rule !== null
    ) {
        if (ruleKey.startsWith('assert/')) {
          logger.warn(
            `The 'assert/' syntax in ${ruleKey} is deprecated. Update your configuration to use 'rule/' instead. Examples and more information: https://redocly.com/docs/cli/rules/configurable-rules/\n`
          );
      }

      const assertion = rule as RawAssertion;
```

This way there's no need to duplicate all the condition.